dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs

```diff
@@ -0,0 +1,320 @@
+    {
+        var properties = new ContainerProperties(collectionName, "/" + collectionName);
+        var response = await this._database.CreateContainerIfNotExistsAsync(properties, cancellationToken: cancel);
+
+        if (response.StatusCode == HttpStatusCode.Created)
+        {
+            this._log.LogInformation("Created collection {0}", collectionName);
+        }
+        else if (response.StatusCode == HttpStatusCode.OK)
+        {
+            this._log.LogInformation("Collection {0} already exists", collectionName);
+        }
+        else
+        {
+            throw new CosmosException("Collection does not exist and was not created", response.StatusCode, 0, collectionName, 0);
+        }
+    }
+
+    /// <inheritdoc />
+    public Task<bool> DoesCollectionExistAsync(string collectionName, CancellationToken cancel = default)
+    {
+        var container = this._database.Client.GetContainer(this._databaseName, collectionName);
+        return Task.FromResult(container != null);
+    }
+
+    /// <inheritdoc />
+    public async Task DeleteCollectionAsync(string collectionName, CancellationToken cancel = default)
+    {
+        var container = this._database.Client.GetContainer(this._databaseName, collectionName);
+        var response = await container.DeleteContainerAsync(cancellationToken: cancel);
+
+        if (response.StatusCode == HttpStatusCode.OK)
+        {
+            this._log.LogInformation("Collection {0} deleted", collectionName);
+        }
+        else
+        {
+            throw new CosmosException("Unable to delete collection", response.StatusCode, 0, collectionName, 0);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<MemoryRecord?> GetAsync(string collectionName, string key, CancellationToken cancel = default)
+    {
+        var container = this._database.Client.GetContainer(this._databaseName, collectionName);
+
+        using (var responseMessage = await container.ReadItemStreamAsync(this.ToCosmosFriendlyId(key), new Microsoft.Azure.Cosmos.PartitionKey(collectionName), cancellationToken: cancel))
+        {
+            if (!responseMessage.IsSuccessStatusCode)
+            {
+                this._log?.LogWarning("Failed to get item {0} from collection {1} with status code {2}", key, collectionName, responseMessage.StatusCode);
+                return null;
+            }
+
+            using (responseMessage.Content)
+            {
+                CosmosDBMemoryRecord record;
+
+                if (typeof(Stream).IsAssignableFrom(typeof(CosmosDBMemoryRecord)))
+                {
+                    record = ((CosmosDBMemoryRecord)(object)responseMessage.Content);
+                }
+                else
+                {
+                    record = await System.Text.Json.JsonSerializer.DeserializeAsync<CosmosDBMemoryRecord>(responseMessage.Content!, cancellationToken: cancel);
```

Comment:
Converting null literal or possible null value to non-nullable type.

Reply:
content non-null is checked in the using statement as well as when the response message succeeds. We can put some extra sauce around this in fit-and-finish. I'll add a tracking task.

Reply:
I'd double request this one as it's a build warning that will impact everyone.

Reply:
will update -standby